utam-compiler/src/main/java/utam/compiler/grammar/UtamElement.java

```diff
@@ -213,7 +209,11 @@ private UtamComposeMethod buildWaitForComposeMethod() {
             null);
     List<UtamMethodAction> compose =
         Collections.singletonList(new UtamMethodActionWaitForElement(name, isLoad()));
-    return new UtamComposeMethod(methodName, description, compose, !isPrivateWait());
+
+    // q: why does this determine the public nature?
+    Boolean isPublic = !isLoad() || isWait();
```

Comment:
wait method should be public only if `wait` is false, we do not need to check !isLoad()